dev-ruby/eventmachine/eventmachine-1.2.7-r1.ebuild

```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+USE_RUBY="ruby25 ruby26 ruby27"
+
+RUBY_FAKEGEM_EXTRADOC="CHANGELOG.md README.md"
+RUBY_FAKEGEM_GEMSPEC="eventmachine.gemspec"
+
+inherit ruby-fakegem
+
+DESCRIPTION="EventMachine is a fast, simple event-processing library for Ruby programs"
+HOMEPAGE="https://github.com/eventmachine/eventmachine"
+SRC_URI="https://github.com/eventmachine/eventmachine/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="|| ( GPL-2 Ruby )"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x64-solaris ~x86-solaris"
+IUSE=""
+
+DEPEND="${DEPEND}
+	dev-libs/openssl:0"
+RDEPEND="${RDEPEND}
+	dev-libs/openssl:0"
+
+ruby_add_bdepend "test? ( dev-ruby/test-unit:2 )"
+
+all_ruby_prepare() {
+	# Remove package tasks to avoid dependency on rake-compiler.
+	rm rakelib/package.rake || die
+
+	sed -i -e '/git ls-files/d' ${RUBY_FAKEGEM_GEMSPEC} || die
+	# Remove the resolver tests since they require network access and
+	# the localhost test fails with an IPv6 localhost.
+	rm tests/test_resolver.rb || die
+
+	# Needs a tty
+	rm tests/test_kb.rb || die
+
+	# Avoid tests that require network access
+	sed -e '/test_bind_connect/,/^  end/ s:^:#:' \
+		-e '/test_invalid_address_bind_connect_src/,/^  end/ s:^:#:' \
+		-e '/test_invalid_address_bind_connect_dst/,/^  end/ s:^:#:' \
+		-i tests/test_basic.rb || die
+	sed -i -e '/test_\(cookie\|http_client\|version_1_0\)/,/^  end/ s:^:#:' \
+		tests/test_httpclient.rb || die
+	sed -i -e '/test_\(get\|https_get\)/,/^  end/ s:^:#:' \
+		tests/test_httpclient2.rb || die
+	sed -i -e '/test_connect_timeout/,/^  end/ s:^:#:' \
+		tests/test_unbind_reason.rb || die
+	sed -i -e '/test_for_real/,/^    end/ s:^:#:' \
+		tests/test_pending_connect_timeout.rb || die
+	# Avoid tests for insecure SSL versions that may not be available
+	sed -e '/test_any_to_v3/,/^    end/ s:^:#:' \
```

Comment:
Yes I validate all the other tests, it was more a first pass to get this working while activate most test,
here more details and recheck:

```sh
# Network needed
$ sed -e '/test_bind_connect/,/^  end/ s:^:#:' \
      -e '/test_invalid_address_bind_connect_src/,/^  end/ s:^:#:' \
      -e '/test_invalid_address_bind_connect_dst/,/^  end/ s:^:#:' \
      -i tests/test_basic.rb || die
Failure: test_invalid_address_bind_connect_dst(TestBasic):
  </unable to resolve address:.*not known/> was expected to be =~
  <"unable to resolve address: Temporary failure in name resolution">.
/var/tmp/portage/dev-ruby/eventmachine-1.2.7-r1/work/ruby25/eventmachine-1.2.7/tests/test_basic.rb:190:in `test_invalid_address_bind_connect_dst'

# works finally, I'd remove this one
$ sed -e '/test_\(cookie\|http_client\|version_1_0\)/,/^  end/ s:^:#:' \
      -i tests/test_httpclient.rb || die

# works finally, I'd remove this one
$ sed -i -e '/test_\(get\|https_get\)/,/^  end/ s:^:#:' \
      tests/test_httpclient2.rb || die

# works finally, I'd remove this one
$ sed -i -e '/test_connect_timeout/,/^  end/ s:^:#:' \
      tests/test_unbind_reason.rb || die

# works finally, I'd remove this one
$ sed -i -e '/test_for_real/,/^    end/ s:^:#:' \
      tests/test_pending_connect_timeout.rb || die

# ssl error, needed
$ sed -e '/test_any_to_v3/,/^    end/ s:^:#:' \
      -e '/test_v3_/,/^    end/ s:^:#:' \
      -e '/test_tlsv1_required_with_external_client/aomit "sslv3"' \
      -e '/test_any_to_any/,/^    end/ s:^:#:' \
      -e '/test_case_insensitivity/,/^    end/ s:^:#:' \
      -e '/test_default_to_default/,/^    end/ s:^:#:' \
      -i tests/test_ssl_protocols.rb || die
Error: test_v3_with_external_client(TestSslProtocols): OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: no protocols available
Error: test_system(TestSystem): RuntimeError: eventmachine not initialized: evma_popen
Error: test_system_with_string(TestSystem): RuntimeError: eventmachine not initialized: evma_popen
Error: test_system_with_string(TestSystem): RuntimeError: eventmachine not initialized: evma_popen
Error: test_dispatch_completion(TestThreadedResource): RuntimeError: eventmachine not initialized: evma_install_oneshot_timer

# Need network
$ sed -e '/test_ipv6_udp_local_server/,/^    end/ s:^:#:' \
      -e '/test_ipv6_tcp_local_server/,/^    end/ s:^:#:' \
      -i tests/test_ipv6.rb || die
unbind: Errno::ECONNREFUSED
Error: test_ipv6_tcp_local_server(TestIPv6): Test::Unit::TestCase::EMTestTimeout: Test was cancelled after 2 seconds.
Error: test_ipv6_udp_local_server(TestIPv6): Test::Unit::TestCase::EMTestTimeout: Test was cancelled after 2 seconds.
```

Reply:
I've looked at this a bit more since I'm working on ruby27 support in the whole tree. My understanding here is that these test break because the actual code is broken on ruby27, so turning off the tests just hides the issues. This is due to ruby27 using a newer openssl implementation. So we can't add ruby27 to this version of eventmachine as-is.

This looks to be fixed upstream and there is recent activity there, so hopefully there will be a new release soon.